test/protocol/test/textDocument/definition/hardhat3/imports.test.ts

```diff
@@ -0,0 +1,178 @@
+      const localFilePath = getProjectPath('hardhat3/contracts/definition/imports/Local1.sol')
+      await client.openDocument(localFilePath)
+
+      const location = await client.findDefinition(toUri(localFilePath), makePosition(7, 5))
+
+      expect(location).to.deep.equal({
+        uri: toUri(getProjectPath('hardhat3/contracts/definition/imports/Local2.sol')),
+        range: makeRange(5, 9, 5, 15),
+      })
+    })
+
+    test('to local file through direct import', async () => {
+      const localFilePath = getProjectPath('hardhat3/contracts/definition/imports/Local1.sol')
+      await client.openDocument(localFilePath)
+
+      const location = await client.findDefinition(toUri(localFilePath), makePosition(8, 5))
+
+      expect(location).to.deep.equal({
+        uri: toUri(getProjectPath('hardhat3/contracts/definition/imports/Local3.sol')),
+        range: makeRange(6, 9, 6, 15),
+      })
+    })
+
+    test('to local file through remapping', async () => {
+      const localFilePath = getProjectPath('hardhat3/contracts/definition/imports/Local1.sol')
+      await client.openDocument(localFilePath)
+
+      const location = await client.findDefinition(toUri(localFilePath), makePosition(9, 5))
+
+      expect(location).to.deep.equal({
+        uri: toUri(getProjectPath('hardhat3/contracts/definition/imports/Local4.sol')),
+        range: makeRange(3, 9, 3, 15),
+      })
+    })
+
+    test('to npm package (without exports) through direct import', async () => {
+      const localFilePath = getProjectPath('hardhat3/contracts/definition/imports/Local2.sol')
+      await client.openDocument(localFilePath)
+
+      const location = await client.findDefinition(toUri(localFilePath), makePosition(6, 2))
+
+      expect(location).to.deep.equal({
+        uri: toUri(getTestPackagePath('pkg_without_exports_1', 'src/A.sol')),
+        range: makeRange(6, 9, 6, 10),
+      })
+    })
+
+    test('to npm package (without exports) through remappings', async () => {
+      const localFilePath = getProjectPath('hardhat3/contracts/definition/imports/Local2.sol')
+      await client.openDocument(localFilePath)
+
+      const location = await client.findDefinition(toUri(localFilePath), makePosition(7, 2))
+
+      expect(location).to.deep.equal({
+        uri: toUri(getTestPackagePath('pkg_without_exports_2', 'src/B.sol')),
+        range: makeRange(3, 9, 3, 10),
+      })
+    })
+
+    test('to npm package (with exports) through direct import', async () => {
+      const localFilePath = getProjectPath('hardhat3/contracts/definition/imports/Local3.sol')
+      await client.openDocument(localFilePath)
+
+      const location = await client.findDefinition(toUri(localFilePath), makePosition(7, 2))
+
+      expect(location).to.deep.equal({
+        uri: toUri(getTestPackagePath('pkg_with_exports_1', 'src/C.sol')),
+        range: makeRange(6, 9, 6, 10),
+      })
+    })
+
+    test('to npm package (with exports) through remappings', async () => {
```

Comment:
Nice. These are clear and will make good regression tests when the HH3 build system changes come through.